src/mito2/src/worker/handle_flush.rs

```diff
@@ -39,6 +40,12 @@ impl<S> RegionWorkerLoop<S> {
         let Some(region) = self.regions.flushable_region_or(region_id, &mut sender) else {
             return;
         };
+        self.update_high_watermark(&region);
+        info!(
+            "Region {} flush request, high watermark: {}",
+            region_id,
+            region.high_watermark_since_flush.load(Ordering::Relaxed)
+        );
```

Comment:
We should update when the flush is finished (e.g. `handle_flush_finished()`). Here we only trigger the flush job.

Reply:
No, it's expected to update the high watermark before going the `schedule_flush` method